SharedPackages/VPN/Sources/VPN/WireGuardKit/WireGuardAdapter.swift

```diff
@@ -128,7 +128,7 @@ private enum State: CustomDebugStringConvertible {
 }
 
 // swiftlint:disable:next type_body_length
-public class WireGuardAdapter {
+public class WireGuardAdapter: WireGuardAdapterProtocol {
```

Comment:
We also have a protocol called `WireGuardInterface` which could be confusing for others looking at this file - they do two different things (this new protocol is for the adapter, whereas the other one abstracts the C interface), but it does make me wonder if `WireGuardInterface` should be renamed for clarity to something like `WireGuardLibraryFFIProtocol` (not a great name, just thinking out loud).

Tbh though we could also just keep this as-is and I don't think any harm would come of it.